src/core/vectortile/qgsvectortilematrixset.h

```diff
@@ -0,0 +1,130 @@
+
+class QgsTileMatrix;
+
+/**
+ * Encapsulates properties of a vector tile matrix set, including tile origins and scaling information.
+ * \ingroup core
+ * \since QGIS 3.22.6
+ */
+class CORE_EXPORT QgsVectorTileMatrixSet : public QgsTileMatrixSet
+{
+  public:
+
+    /**
+     * Returns a vector tile structure corresponding to the standard web mercator/GoogleCRS84Quad setup.
+     */
+    static QgsVectorTileMatrixSet fromWebMercator();
+
+    bool readXml( const QDomElement &element, QgsReadWriteContext &context ) override;
+    QDomElement writeXml( QDomDocument &document, const QgsReadWriteContext &context ) const override;
+
+    /**
+     * Returns the minimum x coordinate for zoom level 0.
+     *
+     * This property is used when scaling raw vector tile coordinates.
+     *
+     * \see setZ0xMinimum()
+     */
+    double z0xMinimum() const { return mZ0xMin; }
+
+    /**
+     * Sets the minimum \a x coordinate for zoom level 0.
+     *
+     * This property is used when scaling raw vector tile coordinates.
+     *
+     * \see z0xMinimum()
+     */
+    void setZ0xMinimum( double x ) { mZ0xMin = x; }
+
+    /**
+     * Returns the maximum x coordinate for zoom level 0.
+     *
+     * This property is used when scaling raw vector tile coordinates.
+     *
+     * \see setZ0xMaximum()
+     */
+    double z0xMaximum() const { return mZ0xMax; }
+
+    /**
+     * Sets the maximum \a x coordinate for zoom level 0.
+     *
+     * This property is used when scaling raw vector tile coordinates.
+     *
+     * \see z0xMaximum()
+     */
+    void setZ0xMaximum( double x ) { mZ0xMax = x; }
+
+    /**
+     * Returns the minimum y coordinate for zoom level 0.
+     *
+     * This property is used when scaling raw vector tile coordinates.
+     *
+     * \see setZ0yMinimum()
+     */
+    double z0yMinimum() const { return mZ0yMin; }
+
+    /**
+     * Sets the minimum \a y coordinate for zoom level 0.
+     *
+     * This property is used when scaling raw vector tile coordinates.
+     *
+     * \see z0yMinimum()
+     */
+    void setZ0yMinimum( double y ) { mZ0yMin = y; }
+
+    /**
+     * Returns the maximum y coordinate for zoom level 0.
+     *
+     * This property is used when scaling raw vector tile coordinates.
+     *
+     * \see setZ0yMaximum()
+     */
+    double z0yMaximum() const { return mZ0yMax; }
+
+    /**
+     * Sets the maximum \a y coordinate for zoom level 0.
+     *
+     * This property is used when scaling raw vector tile coordinates.
+     *
+     * \see z0yMaximum()
+     */
+    void setZ0yMaximum( double y ) { mZ0yMax = y; }
+
+    /**
+     * Initializes the tile structure settings from an ESRI REST VectorTileService \a json map.
+     *
+     * \note This same structure is utilized in ESRI vtpk archives in the root.json file.
+     */
+    bool fromEsriJson( const QVariantMap &json );
+
+  private:
+
+    double mZ0xMin = 0;
+    double mZ0xMax = 0;
+    double mZ0yMin = 0;
+    double mZ0yMax = 0;
```

Comment:
This is where I am confused - why is this necessary to explicitly keep the Z0 rectangle here (rather than accessing extent() from zoom level 0) - is there any difference?

If we need to keep this explicitly, why not use QgsRectangle?

Reply:
Right -- I can confirm that these always exactly match the results from tileMatrix(0).extent(). I've dropped them.